newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -268,17 +270,52 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     }
   }
 
-  _persistValuesToLocalStorage(preferences: Preferences) {
+  _persistValuesToLocalStorage(values: PreferencesValues) {
     try {
-      localStorage.setItem(
-        LocalStorageItem,
-        JSON.stringify(preferences.values)
-      );
+      localStorage.setItem(LocalStorageItem, JSON.stringify(values));
     } catch (e) {
       console.warn('Unable to persist preferences', e);
     }
+  }
+
+  _getLastUsedPath(project: gdProject, kind: ResourceKind) {
+    try {
+      const projectName = project.getName();
+      const values = this.state.values;
+      if (values) {
+        const curProjectPaths = values.lastUsedPath[projectName];
+        if (curProjectPaths) {
+          return curProjectPaths[kind];
+        }
+      }
+    } catch (e) {
+      console.warn('Unable to load latest path', e);
+    }
+  }
+
+  _setLastUsedPath(project: gdProject, kind: ResourceKind, latestPath: string) {
+    try {
+      const projectName = project.getName();
+      const values = this.state.values;
+      if (values) {
+        if (values.lastUsedPath[projectName])
+          values.lastUsedPath[projectName][kind] = latestPath;
+        else {
+          const path: { [ResourceKind]: string } = {};
+          path[kind] = latestPath;
+          values.lastUsedPath[projectName] = path;
+        }
 
-    return preferences;
+        this.setState(
+          {
+            values: values,
```

Comment:
Try to use the spread operator (...) to make the whole function to be almost a single setState :)